training/xgboost/structured/base/trainer/task.py

```diff
@@ -0,0 +1,159 @@
+        # Note: for now, use `cross_val_score` defaults (i.e. 3-fold)
+        scores = model_selection.cross_val_score(estimator, x_val, y_val, cv=3)
+
+        logging.info('Scores: %s', scores)
+
+        metric_output_path = os.path.join(
+            output_dir, 'experiment', metadata.METRIC_FILE_NAME)
+
+        utils.dump_object(scores, metric_output_path)
+
+        # The default name of the metric is training/hptuning/metric.
+        # We recommend that you assign a custom name
+        # The only functional difference is that if you use a custom name,
+        # you must set the hyperparameterMetricTag value in the
+        # HyperparameterSpec object in the job request to match your chosen name
+        hpt = hypertune.HyperTune()
+        hpt.report_hyperparameter_tuning_metric(
+            hyperparameter_metric_tag='my_metric_tag',
+            metric_value=np.mean(scores),
+            global_step=1000)
+
+
+def run_experiment(arguments):
+    """Testbed for running model training and evaluation."""
+    # Get data for training and evaluation
+
+    logging.info('Arguments: %s', arguments)
+
+    dataset = utils.read_df_from_gcs(arguments.input)
+
+    # Get estimator
+    estimator = model.get_estimator(arguments)
+
+    # Run training and evaluation
+    _train_and_evaluate(estimator, dataset, arguments.job_dir)
+
+
+def _parse_args():
+    """Parses command-line arguments."""
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '--log-level',
+        help='Logging level.',
+        choices=[
+            'DEBUG',
+            'ERROR',
+            'FATAL',
+            'INFO',
+            'WARN',
+        ],
+        default='INFO',
+    )
+
+    parser.add_argument(
+        '--input',
+        help='''Dataset to use for training and evaluation.
+              Can be BigQuery table or a file (CSV).
+              If BigQuery table, specify as as PROJECT_ID.DATASET.TABLE_NAME.
+            ''',
+        required=True,
+    )
+
+    parser.add_argument(
+        '--job-dir',
+        help='Output directory for exporting model and other metadata.',
+        required=True,
+    )
+
+    parser.add_argument(
+        '--n-estimators',
+        help='Number of trees in the forest.',
+        default=10,
+        type=int,
+    )
+
+    parser.add_argument(
+        '--max-depth',
+        help='The maximum depth of the tree.',
+        type=int,
+        default=3,
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    """Entry point."""
+
+    arguments = _parse_args()
+    logging.basicConfig(level=arguments.log_level)
+    # Run the train and evaluate experiment
+    time_start = datetime.utcnow()
+    logging.info('Experiment started...')
+    logging.info('.......................................')
+    run_experiment(arguments)
+    time_end = datetime.utcnow()
+    logging.info('.......................................')
+    logging.info('Experiment finished.')
```

Comment:
Is this logging info needed? As all of this is available from the Cloud Console.

Reply:
Good point. Removing it from both xgboost and sklearn samples.